tools/tck-integrity-tests/src/test/scala/org/opencypher/tools/tck/ValidateQuery.scala

```diff
@@ -43,14 +43,16 @@ import scala.util.Success
 import scala.util.Try
 
 trait ValidateQuery extends AppendedClues with Matchers with DescribeStepHelper {
+  private val parsedCache = new scala.collection.concurrent.TrieMap[String, Try[Unit]]()
+  private val normalizedCached = new scala.collection.concurrent.TrieMap[String, Boolean]()
```

Comment:
The general idea is that the integrity tests run in parallel on the scenario level. This does not work at the moment, because some maven extension is not declared thread-safe. In any case, I though it would be good to make this change in a thread-safe manner – Forward looking to a time when these test hopefully will run in parallel.

In general, improving the speed of the integrity tests still needs more work.